src/DefaultUnleash.php

```diff
@@ -59,21 +63,157 @@ public function isEnabled(string $featureName, ?Context $context = null, bool $d
+     * }
+     */
+    public function findFeatureAndDependencies(string $featureName, ?Context $context): ?array
+    {
+        $features = $this->repository->getFeatures();
+        assert(is_array($features));
+        $context ??= $this->configuration->getContextProvider()->getContext();
+
+        $feature = $features[$featureName] ?? null;
+
+        if ($feature === null) {
+            $event = new FeatureToggleNotFoundEvent($context, $featureName);
+            $this->configuration->getEventDispatcherOrNull()?->dispatch(
+                $event,
+                UnleashEvents::FEATURE_TOGGLE_NOT_FOUND,
+            );
+
+            return [
+                'feature' => null,
+                'dependencies' => [],
+            ];
+        }
+
+        $dependencyDefinitions = $feature->getDependencies();
+
+        if ($dependencyDefinitions === null) {
+            return [
+                'feature' => $feature,
+                'dependencies' => [],
+            ];
+        }
+
+        $dependencies = [];
+        foreach ($dependencyDefinitions as $dependencyDefinition) {
+            $name = $dependencyDefinition->getFeature();
+            $dependency = $features[$name] ?? null;
+            if ($dependency !== null) {
+                $dependencies[$name] = $dependency;
+            } else {
+                $event = new FeatureToggleDependencyNotFoundEvent($context, $name);
+                $this->configuration->getEventDispatcherOrNull()?->dispatch(
+                    $event,
+                    UnleashEvents::FEATURE_TOGGLE_NOT_FOUND,
+                );
+            }
+        }
+
+        return [
+            'feature' => $feature,
+            'dependencies' => $dependencies,
+        ];
+    }
+
+    /**
+     * Checks if parent feature flag requirement is satisfied.
+     *
+     * @param Dependency $dependency    the dependency to check
+     * @param Feature    $parentFeature the parent feature to check
+     * @param Context    $context       the context to use
+     */
+    public function isDependencySatisfied(
+        ?Dependency $dependency = null,
+        ?Feature $parentFeature = null,
+        ?Context $context = null,
+    ): bool {
+        if ($dependency === null) {
+            return true;
+        }
+        $context ??= $this->configuration->getContextProvider()->getContext();
+
+        if ($parentFeature === null) {
+            return false;
+        }
+
+        if (count($parentFeature->getDependencies()) > 0) {
+            return false;
+        }
+
+        $parentFeatureEnabled = $this->isFeatureEnabled($parentFeature, $context);
+
+        if ($parentFeatureEnabled->isEnabled() && $dependency->getEnabled() === false) {
+            return false;
+        }
+        if (!$parentFeatureEnabled->isEnabled() && $dependency->getEnabled() !== false) {
+            return false;
+        }
+
+        $dependencyVariants = $dependency->getVariants();
+        if (!empty($dependencyVariants)) {
```

Comment:
Please don't ever use `empty()`, that does like a million things. Be specific, you probably want `count()` instead of `empty()`.

Reply:
Me: *Arrives on this pull request*
Me: *Reads this comment*
Me: Can `empty()` really be that bad?
Me: *Reads for 5 minutes*
Me: Please don't ever use `empty()`, that does like a million things